src/types/context.ts

```diff
@@ -1,11 +1,11 @@
-import { EmitterWebhookEvent as WebhookEvent, EmitterWebhookEventName as WebhookEventName } from "@octokit/webhooks";
+import { EmitterWebhookEvent as WebhookEvent } from "@octokit/webhooks";
 import { Octokit } from "@octokit/rest";
 import { AssistivePricingSettings } from "./plugin-input";
 import { createAdapters } from "../adapters";
 
-export type SupportedEvents = "issues.labeled" | "issues.unlabeled" | "label.edited" | "issue_comment.created";
+export type SupportedEvents = "issues.labeled" | "issues.unlabeled" | "label.edited" | "issue_comment.created" | "push" | "issue_comment"; // last one is needed to satisfy typeguard constraints
```

Comment:
Can you elaborate on this comment?

Reply:
![image](https://github.com/user-attachments/assets/65ba06fa-e307-4e0d-83c1-801b37b22832)


previously T was extending all events so this error wasn't presented but because of the changes below which restricts `Context` to only the supportedEvents I had to include `issue_comment`


```diff
- export interface Context<T extends WebhookEventName = SupportedEvents> {
+ export interface Context<T extends SupportedEvents = SupportedEvents> {
```

Reply:
Seems like a hack but I don't understand the types implementation well enough to offer suggestions. There probably is a way to handle the type definition correctly to resolve this?

Reply:
Previously `T` extended all eventNames and was defaulted to be the `SupportedEvents` union which meant the `Context` generic could be supplied with `issue_comment` without error but also meant that any other eventName could be passed in without error, which you'd expect for it to throw if it's not a member of `SupportedEvents`

---

Opted for this below so that `issue_comment` is not a member of `SupportedEvents` but can still be used as a generic

```ts
export interface Context<T extends SupportedEvents | "issue_comment" = SupportedEvents> {
```

Reply:
I am not great with type definitions, but seems like you did your research!